### DISCUSSION TITLE
Knapp for ekspandering av innhold

### DISCUSSION THREAD

**Discussion by piofinn** [fremtind/jokul#2393] [Annet]
I det siste har jeg begynt å legge merke til et lite mønster som brukes en del steder, men som ikke har noen god, enhetlig løsning: Visning og skjuling av innhold.

Vi har jo et generelt prinsipp om at vi vil skjule så lite informasjon som mulig fra brukerne våre, men noen ganger er det likevel nødvendig eller fornuftig å kunne skjule større mengder med ikke-superviktig informasjon. Accordion er fint til FAQ-varianter, men føles litt rar å bruke frittstående med bare ett element, og den løser heller ikke ting som ekspanderende tabellrader.

Jeg foreslår derfor en liten knappekomponent for å vise/skjule innhold. Noe á la `ExpandButton` og `InlineExpandButton` [i FFE](https://design.sparebank1.no/komponenter/knapper/#expandbutton). Under er noen eksempler på dette mønstret fra både eksisterende løsninger og skisser:

![endre](https://user-images.githubusercontent.com/25739615/137867109-890be24a-137c-458c-bc02-abb01c892169.gif)

<img width="340" alt="visforskjellerlukket" src="https://user-images.githubusercontent.com/25739615/137867203-6c4b2adc-ab09-47cc-a882-e7a125f9d617.png">
<img width="364" alt="visforskjelleråpen" src="https://user-images.githubusercontent.com/25739615/137867205-21bfe2f6-40f4-47b5-829b-163ea5d78e1d.png">

<img width="497" alt="ansattlukket" src="https://user-images.githubusercontent.com/25739615/137867376-9e836a96-5cfd-4cbf-8a3f-ca5d4f171ddf.png">
<img width="510" alt="ansattåpen" src="https://user-images.githubusercontent.com/25739615/137867372-7f2cfdf8-3af4-4abd-a41b-e47be6669a39.png">

![viskode](https://user-images.githubusercontent.com/25739615/137867484-916edaa6-4fc8-4042-bd2a-5194371570be.gif)

**Comment by wkillerud** [comment]
Det vil kunne bli et behov i tabeller også for ekspanderbare rader. Det gir kanskje mening å ha labelen som valgfri?

**Reply by piofinn** [reply]
Den første gifen er faktisk fra en ekspanderende tabellrad! Og ja, tenker også at label kan være valgfri for de tilfellene 😊

**Comment by ivnai** [comment]
(Er her gode plass for å skrive eller bør man først diskutere i design forum) Ser behov for en slik komponent og bruker denne på BSF. Er egentlig delen av accordion, trodde er mer konsekvent da pilen står først.
![Screenshot 2021-11-19 at 14 32 23](https://user-images.githubusercontent.com/93194579/142631903-e9ceb18a-5101-4c72-b31b-ceb688ca6b05.png)

**Reply by wkillerud** [reply]
Supert med eksempler og å si ifra at man trenger noe her på Discussions @ivnai 👍

Jeg husker ikke når, men vi diskuterte på et designsystemforum om pilen burde stå først eller etter teksten. Det var om[ ekspanderbare tabeller](https://jokul.fremtind.no/komponenter/table#ekspanderende-tabeller). Da ble vi enige om at pilen bør komme sist (og at _kanskje_ accordion burde endres?). Så nye expand buttons burde nok følge det siste, med pil på høyre.

**Comment by wkillerud** [comment]
Bør vi ha noen føringer for hvordan innholdet vises og skjules? Eller er det bare snakk om her å lage en komponent som har en animert label og pil (`Tekst 🔽`), så er selve ekspanderingen/visningen/skjulingen opp til den enkelte designer og utvikler?

**Reply by ivnai** [reply]
Sikkert lurt å sjekke føringer ja. Hvordan tar vi dette videre? Fint å inkludere flere designere.. bør jeg ta dette til VD forum først?

**Reply by wkillerud** [reply]
Vi kan snakke om det på Designsystemforum i morgen, men om du vil ta det til VD-forum også er det supert @ivnai 👍 Fint med tilbakemeldinger fra flere.

Når vi blir enige om design kan en utvikler fra ett av teamene som trenger det kode komponenten. Jeg hjelper gjerne til med det praktiske, som å lage riktig kodestruktur for pakker og dokumentasjon på jokul.fremtind.no.

**Comment by ivnai** [comment]
Smartforhandler
![Screenshot 2021-11-23 at 13 02 13](https://user-images.githubusercontent.com/93194579/143020935-4d63f3f2-a84f-4714-92d1-90ee8d0c4ea1.png)
![Screenshot 2021-11-23 at 13 02 30](https://user-images.githubusercontent.com/93194579/143020938-e12ea3b5-92fd-4a9d-b9b8-481358a32664.png)

Felles PM - Oversiktssider 
![Screenshot 2021-11-23 at 13 05 40](https://user-images.githubusercontent.com/93194579/143021473-c8929c94-1de8-49c8-8ac9-7aab2f078fcf.png)

Felles PM - Kjøpsflyt
![Screenshot 2021-11-23 at 13 10 44](https://user-images.githubusercontent.com/93194579/143021935-668850c5-8dda-4566-ac6c-e50dc157c4fa.png)

BM - Tilganger
![Screenshot 2021-11-23 at 13 33 21](https://user-images.githubusercontent.com/93194579/143024701-2797ace6-4650-4a2b-b8bf-1aa7fbf8aff7.png)

**Comment by wkillerud** [comment]
### Oppsummert fra forum 23.11

Gruppe bestående av @espkva @Stakset @ivnai @AudunBerdal @wkillerud ble enige om å lage en ny komponent. 

@espkva sitter med en versjon i app som kan løftes inn i Jøkul. @wkillerud hjelper med det praktiske.

Kort fortalt går vi for en førsteversjon som likner litt på en [accordion item](https://jokul.fremtind.no/komponenter/accordion), det vil si samme pil og animasjoner av den, pluss en tekst som blir bold når knappen er ekspandert. Pilen blir i motsetning til en Accordion på _høyre_ side av teksten, og inline (rett etter teksten).

**Comment by wkillerud** [comment]
## :shipit: Release

Da er første versjon ute! Takk til alle involverte 👏

- [Dokumentasjon](https://jokul.fremtind.no/komponenter/expandbutton)
- [Figma](https://www.figma.com/file/y9JtTf3whfTCHCdYSUXVQ5/?node-id=62%3A3085)
- [Hvordan installere](https://github.com/fremtind/jokul/tree/main/packages/expand-button-react#fremtindjkl-expand-button-react)
